Sources/Networking/Core/DownloadAPIManaging.swift

```diff
@@ -13,7 +13,7 @@ public typealias DownloadResult = (URLSessionDownloadTask, Response)
 /// A definition of an API layer with methods for handling data downloading.
 /// Recommended to be used as singleton.
 /// If you wish to use multiple instances, make sure you manually invalidate url session by calling the `invalidateSession` method.
-public protocol DownloadAPIManaging {
+public protocol DownloadAPIManaging: Actor {
```

Comment:
@matejmolnar before I'll dive deeper into the whole code, I'm not sure I'm sold on this 😄 . 

Did you also consider marking the functions `async` instead? I don't know what is the best solution yet, I'm just little hesitant on marking the protocol as `Actor`, so just wondering whether you thought about the other solution as well and maybe why you decided to go this way? ☺️